File: py/leetcode-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        chardict = {}
        for ch in s:
            try:
                chardict[ch] += 1
            except:
                chardict[ch] = 1
        

        for ch in t:
            try:
                chardict[ch] -= 1
                if chardict[ch] < 0:
                    return False
            except:
                return False
        
        for key in chardict.keys():
            if chardict[key] != 0:
                return False
        
        return True
            

File: rs/leetcode-anagram.rs
use std::collections::HashMap;
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        let mut char_count: HashMap<char, i32> = HashMap::new();

        for ch in s.chars() {
            *char_count.entry(ch).or_insert(0) += 1;
        }

        for ch in t.chars() {
            if let Some(count) = char_count.get_mut(&ch) {
                *count -= 1;
                if *count < 0 {
                    return false;
                }
            } else {
                return false;
            }
        }

        char_count.values().all(|&count| count == 0)
    }
}